### DISCUSSION TITLE
x-trap fails to focus on modal when element opened by an actual mouse click

### DISCUSSION THREAD

**Discussion by DaTrader** [alpinejs/alpine#3186] [7. Bugs (archived)]
I am using Alpine with Phoenix LiveView and I have a button on my page that opens a modal with a form (makes LiveView render it on click) . I use x-trap to trap keyboard navigation within the modal and I have noticed the following behavior:

a) If the button is "clicked" by pressing enter, the modal form opens, tab/shift tab navigation works fine and when I press space or enter or on an input such as a radio button or a button of type button, the events get processed properly and everything works;

b) However, if the button that opens the form modal is actually clicked on (with the mouse), once it opens the modal, although the tab navigation works fine and the focus is moved from one element to another, when pressing space or enter on those inputs, the modal closes. It closes because the modal's LiveView phx-click-away is set to close it (if removed, nothing happens, and neither does the desired input event). I've noticed that this happens because x-trap does not focus (as in programmatically "clicks on") the form. I know this, because if I manually click on the form before pressing space or enter on one of its inputs, the inputs will work. Accordingly, my workaround here is to simulate the click my self in the x-init, as shown below:

JS:

```
window.simulateClick = function( targetNode) {
  targetNode.dispatchEvent( new Event( "mouseover", { bubbles: true, cancelable: false}));
  targetNode.dispatchEvent( new Event( "mousedown", { bubbles: true, cancelable: false}));
  targetNode.dispatchEvent( new Event( "mouseup", { bubbles: true, cancelable: false}));
  targetNode.dispatchEvent( new Event( "click", { bubbles: true, cancelable: false}));
};

```
LiveView template:

```
      ..
        <div
          id="modal-content"
          class="animate-fade-in-scale max-h-full overflow-auto"
          phx-click-away={hide_modal(@close_modal_target)}
          phx-window-keydown={hide_modal(@close_modal_target)}
          phx-key="escape"
          x-init="$nextTick(() => { simulateClick( $el) })"
          x-trap="true"
       >
         ...

```
It is worth noting that dispatching just a "click" event instead doesn't work and neither does doing setting x-init as follows: `x-init="$focus.focus( $el)`.

**Comment by ekwoka** [comment]
I think easier than doing a simulated click would be to set it to a state that is false and then in the nextTick set that state to true.

Curious though 🤔

**Reply by DaTrader** [reply]
```
          x-data="{ open: false }"
          x-init="$nextTick(() => { open = true })"
          x-trap="open"
```

If this is what you're referring to, it doesn't work either. Pressing enter on a (non-submit) button in the form closes it, for neither the form nor its parent element (this div) are focused by an actual (or simulated) mouse click.
